grails-app/views/order/_summary.gsp

```diff
@@ -171,14 +171,22 @@
                                 ${warehouse.message(code: 'default.button.rollback.label')}
                             </g:link>
                             <g:if test="${!orderInstance?.hasInvoice && orderInstance?.paymentTerm?.prepaymentPercent &&
```

Comment:
We should encapsulate this logic in a domain method. In fact, I think we use each component in other places so let's add methods for each of them.
```
Boolean isPrepaymentInvoiceAllowed() { 
    return !this.hasInvoice && this.isPrepaymentRequired && !this.empty()
}

Boolean isPrepaymentRequired() { 
    return this.paymentTerm?.prepaymentPercent > 0
}

Boolean isEmpty() { 
    return this.orderItems?.empty && orderInstance?.orderAdjustments?.empty
}
```

I can't remember the convention for is/has methods in Grails, so you might have to play around with isSomeCondition() vs getIsSomeCondition().

And isEmpty is not descriptive enough so please suggest a more descriptive name if you can think of one. The following is a little long and awkward, but descriptive.
```
isOrderItemsAndOrderAdjustmentsEmpty
```

Reply:
I'll tackle it with refactors for Order.getInvoiceItems